Source/processcontainers/implementations/CRunImplementation/CRunImplementation.cpp

```diff
@@ -0,0 +1,460 @@
+#include "CRunImplementation.h"
+#include "JSON.h"
+#include <thread>
+
+namespace WPEFramework
+{    
+namespace ProcessContainers
+{
+    // NetworkInterfaceIterator
+    // ----------------------------------
+    CRunNetworkInterfaceIterator::CRunNetworkInterfaceIterator()
+        : NetworkInterfaceIterator()
+    {
+        TRACE_L1("CRunNetworkInterfaceIterator::CRunNetworkInterfaceIterator() not implemented")
+    }
+
+    CRunNetworkInterfaceIterator::~CRunNetworkInterfaceIterator()
+    {
+        TRACE_L1("CRunNetworkInterfaceIterator::~CRunNetworkInterfaceIterator() not implemented")
+    }
+
+    std::string CRunNetworkInterfaceIterator::Name() const 
+    {
+        TRACE_L1("CRunNetworkInterfaceIterator::Name() not implemented")
+
+        return "";
+    }
+
+    uint32_t CRunNetworkInterfaceIterator::NumIPs() const 
+    {
+        TRACE_L1("CRunNetworkInterfaceIterator::NumIPs() not implemented")
+
+        return 0;
+    }
+
+    std::string CRunNetworkInterfaceIterator::IP(uint32_t id) const 
+    {
+        TRACE_L1("CRunNetworkInterfaceIterator::IP() not implemented")
+
+        return "";
+    }
+
+    // Container administrator
+    // ----------------------------------
+    IContainerAdministrator& ProcessContainers::IContainerAdministrator::Instance()
+    {
+        static CRunContainerAdministrator& runCContainerAdministrator = Core::SingletonType<CRunContainerAdministrator>::Instance();
+
+        return runCContainerAdministrator;
+    }
+
+    IContainer* CRunContainerAdministrator::Container(const string& id, IStringIterator& searchpaths,  const string& logpath, const string& configuration) 
+    {
+        while (searchpaths.Next()) {
+            auto path = searchpaths.Current();
+
+            Core::File configFile(path + "/Container/config.json");
+
+            if (configFile.Exists()) {
+                CRunContainer* container = new CRunContainer(id, path + "/Container", logpath);
+                _containers.push_back(container);
+                AddRef();
+
+                return container;
+            }
+        }
+
+        return nullptr;
+    }
+
+    CRunContainerAdministrator::CRunContainerAdministrator()
+        : _refCount(1)
+    {
+
+    }
+
+    CRunContainerAdministrator::~CRunContainerAdministrator()
+    {
+        if (_containers.size() > 0) {
```

Comment:
Since it is a singleton, lets make sure that there is only 1 reference left if we are destructed. The reference we hold on our selves. So assert (_refCount == 1) might be interesting here. Than you will see that adding the reference count (addRef on the instance, will balance the lifetime managament correctly.